### PR TITLE
Sphinx docs

### PR DESCRIPTION
**Pull request by tino**
I think there is quite some value in the old but quite extensive lamson docs, so I started porting them (they are from before the licensing trouble) in this branch. I setup Sphinx as it is the defacto python documentation tool, and can easily be deployed to readthedocs.org (instead of creating a salmonproject.org as you mentioned in doc/README).

So far I converted the index and getting_started. I also tried getting sphinx's autogen/summary to work, but that is not working completely yet.

What do you think, should I continue converting and purging these docs, and put them up on rtd?

~~Btw. the name "salmon" is already forgiven on both pypi and rtd, so what to deploy this under? salmon-mail?~~ I see you already settled for salmon-mail :+1:

### CODE REVIEW

**Comment by moggers87** [comment]
See discussion on https://github.com/moggers87/salmon/issues/27

tl;dr: the old docs weren't licensed under anything (they weren't part of the source code) and I'd like to go for home-grown documentation that's properly licensed

**Comment by tino** [comment]
Pff, I guess I need to check closed issues better.

Anyway, using this is quite hard without any docs, so I hope you get something up...

**Comment by tino** [comment]
I guess starting with a sphinx project with the apidocs should work right? They are part of the code, license wise and "update requiring"-wise. Gives at least some way to easily browse the parts.

I'll wait for some feedback before I set it up.

**Comment by moggers87** [comment]
You should probably talk to @tflink to avoid duplication of effort :)
